src/controllers/accommodationsControllers.js

```diff
@@ -0,0 +1,170 @@
+  }
+  if(req.body.images_links){
+    facility.images_links = (req.body.image_links);
+  }
+  if(req.body.add_on_services){
+    facility.add_on_services = (req.body.add_on_services);
+  }
+  if(req.body.amenities){
+    facility.amenities = (req.body.amenities);
+  }
+
+  let result = await AccommodationService.updateOne(req.params.id,facility);
+  result = JSON.parse(JSON.stringify(result));
+  return successResponse(res, 201, "Accommodation facility updated successfully", result);;
+};
+
+accommodationController.deleteOne = async (req, res) => {
+  const facility = await AccommodationService.getOne(req.params.id);
+  if(!(facility)){
+    return errorResponse(res, 404, "Accommodation facility not found!");
+  }
+  const result = await AccommodationService.deleteOne(req.params.id);
+  const roomDeletion = await AccommodationService.deleteAllRooms(req.params.id);
+  return successResponse(res, 200, "Accommodation facility deleted successfully", {accommodation_deletions: result, room_deletions : roomDeletion});
+};
+
+
+//////////////////////////////////////////////////////////////////////////
+/////////////////////////////////////////////////////////////////////////
+
+accommodationController.getAllRooms = async (req, res) => {
+  const accommodationId = req.params.id;
+  let result = await AccommodationService.getAllRooms(accommodationId);
+  result = JSON.parse(JSON.stringify(result));
+  return successResponse(res, 200, "Rooms querried from the DB successfully", result);;
+};
+
+accommodationController.createOneRoom = async (req, res) => {
+  const accommodationId = req.params.id;
+  let facility = await AccommodationService.getOne(accommodationId);
+  if(!(facility)){
+    return errorResponse(res, 404, "Accommodation facility not found!");
+  }
+  req.body.accommodation_id = accommodationId;
+  let result = await AccommodationService.createOneRoom(req.body);
+  result = JSON.parse(JSON.stringify(result));
+  return successResponse(res, 201, "Room created in the DB successfully", result);;
+};
+
+accommodationController.updateOneRoom = async (req, res) => {
+  
+  let room = await AccommodationService.getOneRoom(req.params.id,req.params.roomId);
+  if(!(room)){
+    return errorResponse(res, 404, `Room with id = ${req.params.roomId} from accommodation facility of id = ${req.params.id}  not found!`);
+  }
+  room = JSON.parse(JSON.stringify(room));
+  if(req.body.accommodation_id){
+    const facility = await AccommodationService.getOne(req.body.accommodation_id);
+    if(!(facility)){
+      return errorResponse(res, 404, "Accommodation facility not found!");
+    }
+    room.accommodation_id = req.body.accommodation_id;
+  }
+  
+  if(req.body.bed_type){
+    room.bed_type =req.body.bed_type;
+  }
+  if(req.body.cost){
+    room.cost =req.body.cost;
+  }
+  if(req.body.image_link){
+    room.images_links.push(req.body.image_link);
+  }
+  if(req.body.images_links){
+    room.images_links = (req.body.images_links);
+  }
```

Comment:
what if any field above doesn't have the value what will happen

Reply:
If the field does not have a value, then it's value will be evaluated to `undefined`, and it's corresponding if-statement won't run & the corresponding field will not be updated in the DB.